packages/headless/src/api/knowledge/knowledge-api-answer.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  EventSourceMessage,
+  fetchEventSource,
+} from '@microsoft/fetch-event-source';
+import {GeneratedContentFormat} from '../../features/generated-answer/generated-response-format';
+import {selectQuery} from '../../features/query/query-selectors';
+import {QueryState} from '../../features/query/query-state';
+import {selectSearchHub} from '../../features/search-hub/search-hub-selectors';
+import {ConfigurationSection} from '../../state/state-sections';
+import {GeneratedAnswerCitation} from '../generated-answer/generated-answer-event-payload';
+import {SearchRequest} from '../search/search/search-request';
+import {knowledgeApi} from './knowledge-api';
+
+type StateNeededByKnowledgeAPI = ConfigurationSection;
+
+interface ErrorPayload {
+  message?: string;
+  code?: number;
+}
+
+class FatalError extends Error {
+  constructor(public payload: ErrorPayload) {
+    super(payload.message);
+  }
+}
+
+interface GeneratedAnswerStream {
+  answerStyle: string | undefined;
+  contentFormat: GeneratedContentFormat | undefined;
+  answer: string | undefined;
+  citations: GeneratedAnswerCitation[] | undefined;
+  generated: boolean;
+  isStreaming: boolean;
+  isLoading: boolean;
+}
+
+interface Data {
+  answerStyle: string | undefined;
+  contentFormat: GeneratedContentFormat | undefined;
+  answer: string | undefined;
+  citations: GeneratedAnswerCitation[] | undefined;
+  generated: boolean;
+  isStreaming: boolean;
+  isLoading: boolean;
+}
+
+const updateCacheWithEvent = (event: EventSourceMessage, draft: Data) => {
+  const message = JSON.parse(event.data);
+  const payload = JSON.parse(message.payload);
+  if (message.payloadType === 'genqa.headerMessageType') {
+    draft.answerStyle = payload.answerStyle;
+    draft.contentFormat = payload.contentFormat;
+    draft.isStreaming = true;
+    draft.isLoading = false;
+  }
+
+  if (message.payloadType === 'genqa.messageType') {
+    if (draft.answer === undefined) {
+      draft.answer = payload.textDelta;
+    } else {
+      draft.answer = draft.answer?.concat(payload.textDelta);
+    }
+  }
+
+  if (message.payloadType === 'genqa.citationsType') {
+    draft.citations = payload.citations;
+  }
+
+  if (message.payloadType === 'genqa.endOfStreamType') {
+    draft.generated = Boolean(payload.answerGenerated);
+    draft.isStreaming = false;
+  }
```

Comment:
I always prefered `If` to switchcases, thats why I do not have the reflex to use them. But since we better `return` at each of them, I will use a switch case to `break`